LECT_10 (Google Storage)/dags/gcloud.py

```diff
@@ -0,0 +1,105 @@
+"""
+Upload a file to GCS
+
+Notes:
+    - all examples can be found here:
+    https://github.com/googleapis/python-storage/tree/main/samples/snippets
+"""
+from google.cloud import storage
+from datetime import datetime
+
+from airflow import DAG
+from airflow.operators.empty import EmptyOperator
+from airflow.operators.python import PythonOperator
+
+from google.oauth2 import service_account
+
+import json
+import os
+import tempfile
+
+def upload_blob(bucket_name, source_file_path, destination_blob_name):
+    """
+    Uploads a file to the bucket.
+    """
+    # The ID of your GCS bucket
+    # bucket_name = "your-bucket-name"
+    # The path to your file to upload
+    # source_file_name = "local/path/to/file"
+    # The ID of your GCS object
+    # destination_blob_name = "storage-object-name"
+    
+    storage_client = storage.Client.from_service_account_json('/opt/airflow/keys/my-project-my-marks-7b258129da57.json')
+    
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(destination_blob_name)
+
+    blob.upload_from_filename(source_file_path)
+
+    print(
+        f"File {source_file_path} uploaded to {destination_blob_name}."
+    )
+
+
+
+
+DEFAULT_ARGS = {
+    'depends_on_past': False,
+    'email': ['admin@example.com'],
+    'email_on_failure': True,
+    'email_on_retry': False,
+    'retries': 2,
+    'retry_delay': 20,
+}
+
+def run_job1():
+	upload_blob(
+		bucket_name='de2022-paul_skyba_bucket_1',
+		source_file_path='/opt/airflow/gcloud/src1_sales_2022-08-01__01.csv',
+		destination_blob_name='src1/sales/v1/2022/08/01/src1_sales_2022-08-01__01.csv',
+	)
+
+
+def run_job2():
+	upload_blob(
+		bucket_name='de2022-paul_skyba_bucket_1',
+		source_file_path='/opt/airflow/gcloud/src1_sales_2022-08-02__01.csv',
+		destination_blob_name='src1/sales/v1/2022/08/02/src1_sales_2022-08-02__01.csv',
```

Comment:
dates in the path should be generated dynamically by Airflow on the run time.
IOW you should use a templated variable`{{ dag_run.logical_date }}`. More details here: https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html